scripts/git-hooks/pre-commit.sh

```diff
@@ -11,6 +16,34 @@ if [ $RESULT -ne 0 ]; then
   ./gradlew spotlessApply > /dev/null
 
   echo "Recommended changes from spotless have been applied."
+  exit $RESULT
+fi
+
+##################
+# gitleaks
+##################
+
+# Use sentry.properties file as a proxy for determining if the contributor has
+# access to secrets and should run gitleaks before committing
```

Comment:
Thanks for testing this out Phil!

> I currently don't have sentry.properties file locally

That's a really good sign that I didn't pick a good check here then! 😄 

> how about we only run the gitleaks check if you have the command installed? 

I just don't want someone with access to Pocket Casts's secrets to be able to forget to install gitleaks. With that said, I initially went with your approach because I couldn't think of a better check. Then I realized that I could just check for the existence of `~/.mobile-secrets`, which I think should be a pretty solid check. 

If for some reason that doesn't work, we can revert the last two commits on this PR and that gives us a setup that doesn't run gitleaks if it isn't installed but it gives lots of warning messages in that case. Let me know what you think.

Reply:
Thanks, that looks great.